Source/Methods/RFmxDPD/RFmxDPD.cs

```diff
@@ -97,15 +168,72 @@ public struct MemoryPolynomialResults
         #endregion
 
         #region ConfigureDPD
+        public static Waveform ConfigurePreDPDCrestFactorReduction(RFmxSpecAnMX specAn, Waveform referenceWaveform, PreDpdCrestFactorReduction preDpdCfr, string selectorString = "")
+        {
+            Waveform preDpdCfrWaveform = referenceWaveform;
+            if (preDpdCfr.Enabled == RFmxSpecAnMXDpdPreDpdCfrEnabled.True)
+            {
+                // Configure the new waveform
+                preDpdCfrWaveform.Data = referenceWaveform.Data.Clone(); // clone waveform so RFmx can't act on reference waveform
+                preDpdCfrWaveform.UpdateWaveformNameAndScript(referenceWaveform.Name + "preDPDCFR");
+
+                //Configure Pre-DPD CFR             
+                RFmxSpecAnMXDpdApplyDpdIdleDurationPresent preDpdCfrIdlePresent = preDpdCfrWaveform.IdleDurationPresent ?
+                    RFmxSpecAnMXDpdApplyDpdIdleDurationPresent.True : RFmxSpecAnMXDpdApplyDpdIdleDurationPresent.False;
+                specAn.Dpd.PreDpd.SetCfrEnabled(selectorString, preDpdCfr.Enabled);
+                specAn.Dpd.PreDpd.SetCfrMethod(selectorString, preDpdCfr.Method);
+                specAn.Dpd.PreDpd.SetCfrMaximumIterations(selectorString, preDpdCfr.MaxIterations);
+                specAn.Dpd.PreDpd.SetCfrTargetPapr(selectorString, preDpdCfr.TargetPapr_dB);
+                specAn.Dpd.PreDpd.SetCfrWindowType(selectorString, preDpdCfr.WindowType);
+                specAn.Dpd.PreDpd.SetCfrWindowLength(selectorString, preDpdCfr.WindowLength);
+                specAn.Dpd.PreDpd.SetCfrShapingFactor(selectorString, preDpdCfr.ShapingFactor);
+                specAn.Dpd.PreDpd.SetCfrShapingThreshold(selectorString, preDpdCfr.ShapingThreshold_dB);
+                specAn.Dpd.PreDpd.SetCfrFilterEnabled(selectorString, preDpdCfr.FilterEnabled);
+                specAn.Dpd.PreDpd.SetCfrNumberOfCarriers(selectorString, preDpdCfr.CarrierChannels.Length);
+
+                string carrierString;
+                for (int i = 0; i < preDpdCfr.CarrierChannels.Length; i++)
+                {
+                    carrierString = RFmxSpecAnMX.BuildCarrierString2(selectorString, i);

```

Comment:
move carrier string declaration into the loop. i don't see it used anywhere else.

string carrierString = RFmxSpecAnMX.BuildCarrierString2(selectorString, i);

Reply:
But why declare it as a new variable each iteration of the loop & allocate memory? Should it not be declared only once? Doesn't that (theoretically) increase overhead?

Reply:
variables declared inside a loop aren't allocated every loop iteration.

Reply:
6eb090a0facfdb52295d84dd783f3472f2487408